nodejs/src/index.js

```diff
@@ -22,9 +26,18 @@ exports.write = function (struct, compression, callback) {
 };
 
 exports.read = function (Model, rawData, callback) {
+	var callback = arguments.length === 4 ? arguments[3] : arguments[2];
```

Comment:
I don't like `.pop` on arguments, but `arguments[arguments.length-1]` is fine by me, they're both readable.

Yes, the arguments shuffling is for backward compatibility and for the laziness of not specifying a parameter when compression is inferred from the bytes